Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/AnalyticsConfigMapper.kt

```diff
@@ -4,20 +4,59 @@ import dagger.Reusable
 import de.rki.coronawarnapp.appconfig.AnalyticsConfig
 import de.rki.coronawarnapp.appconfig.SafetyNetRequirements
 import de.rki.coronawarnapp.appconfig.SafetyNetRequirementsContainer
+import de.rki.coronawarnapp.appconfig.internal.ApplicationConfigurationInvalidException
 import de.rki.coronawarnapp.server.protocols.internal.v2.AppConfigAndroid
 import javax.inject.Inject
 
 @Reusable
 class AnalyticsConfigMapper @Inject constructor() : AnalyticsConfig.Mapper {
     override fun map(rawConfig: AppConfigAndroid.ApplicationConfigurationAndroid): AnalyticsConfig {
-        // TODO
-        return AnalyticsConfigContainer()
+        if (!rawConfig.hasPrivacyPreservingAnalyticsParameters() &&
+            !rawConfig.privacyPreservingAnalyticsParameters.hasCommon() &&
+            !rawConfig.privacyPreservingAnalyticsParameters.hasPpac()
+        ) {
+            throw ApplicationConfigurationInvalidException(
+                message = "Analytics Parameters are missing"
+            )
+        }
+
+        return rawConfig.mapAnalyticsConfig()
     }
 
+    private fun AppConfigAndroid.ApplicationConfigurationAndroid.mapSafetyNet(): SafetyNetRequirementsContainer =
+        this.privacyPreservingAnalyticsParameters.ppac.let {
+            SafetyNetRequirementsContainer(
+                requireBasicIntegrity = it
+                    .requireBasicIntegrity,
+                requireCTSProfileMatch = it
+                    .requireCTSProfileMatch,
+                requireEvaluationTypeBasic = it
+                    .requireEvaluationTypeBasic,
+                requireEvaluationTypeHardwareBacked = it
+                    .requireEvaluationTypeHardwareBacked
+            )
+        }
+
+    private fun AppConfigAndroid.ApplicationConfigurationAndroid.mapAnalyticsConfig(): AnalyticsConfigContainer =
+        this.privacyPreservingAnalyticsParameters.common.let {
+            AnalyticsConfigContainer(
+                safetyNetRequirements = this.mapSafetyNet(),
+                probabilityToSubmit = it
+                    .probabilityToSubmit,
+                hoursSinceTestResultToSubmitKeySubmissionMetadata = it
+                    .hoursSinceTestResultToSubmitKeySubmissionMetadata,
+                hoursSinceTestRegistrationToSubmitTestResultMetadata = it
+                    .hoursSinceTestRegistrationToSubmitTestResultMetadata,
+                probabilityToSubmitNewExposureWindows = it
+                    .probabilityToSubmitExposureWindows
+            )
+        }
+
     data class AnalyticsConfigContainer(
         override val safetyNetRequirements: SafetyNetRequirements = SafetyNetRequirementsContainer(),
-        override val probabilityToSubmit: Float = 1f,
-        override val probabilityToSubmitAfterRiskCalculation: Float = 1f,
-        override val probabilityToSubmitNewExposureWindows: Float = 1f
+        override val probabilityToSubmit: Double = 1.0,
```

Comment:
Should the defaults be 1.0, or would it be better if the defaults were 0.0, i.e. disabled? Does the spec say?

Reply:
not really, i'll remove the defaults for now as there is no way to instantiate the defaults anyways at the moment